src/main/java/seedu/address/model/UniqueRoomList.java

```diff
@@ -0,0 +1,355 @@
+        requireAllNonNull(readOnlyRoomList);
+
+        ObservableList<Room> roomLists = readOnlyRoomList.getRoomObservableList();
+        rooms.addAll(roomLists);
+        internalList.addAll(roomLists);
+    }
+
+    public void setData(Room room) {
+        int roomNumber = room.getRoomNumber();
+        if (room.getRoomNumber() > internalList.size()) {
+            for (int i = internalList.size(); i < roomNumber - 1; i++) {
+                Room roomToAdd = new Room(i + 1);
+                internalList.add(roomToAdd);
+                rooms.add(room);
+            }
+            rooms.add(room);
+            internalList.add(room);
+        } else {
+            Room currRoom = internalList.get(roomNumber - 1);
+            internalList.remove(roomNumber - 1);
+            rooms.remove(currRoom);
+            internalList.add(roomNumber, room);
+            rooms.add(room);
+        }
+    }
+    /**
+     * Returns Priority Queue of rooms
+     */
+    public PriorityQueue<Room> getRooms() {
+        return this.rooms;
+    }
+
+    /**
+     * Returns number of rooms in hotel
+     */
+    public int getNumOfRooms() {
+        return internalList.size();
+    }
+
+    public ObservableList<Room> getRoomObservableList() {
+        return internalList;
+    }
+
+    /**
+     * Adds this room to the RoomList
+     * @param room is added to RoomList
+     */
+    public void addRooms(Room room) {
+        this.numOfRooms++;
+        rooms.add(room);
+        internalList.add(room);
+    }
+
+    /**
+     * Adds the number of the rooms in a hotel
+     *
+     * @param numOfRooms is the number of rooms to be added
+     */
+    public void addRooms(int numOfRooms) {
+        this.numOfRooms = numOfRooms;
+        addRooms();
+    }
+
+    private void addRooms() {
+        if (numOfRooms <= 0) {
+            return;
+        }
+        if (numOfRooms > internalList.size()) {
+            for (int i = internalList.size(); i < numOfRooms; i++) {
+                Room room = new Room(i + 1);
+                rooms.add(room);
+                internalList.add(room);
+            }
+        } else if (numOfRooms < internalList.size()) {
+            List<Room> occupiedRooms = occupiedRooms();
+            List<Room> unoccupiedRooms = unOccupiedRooms();
+            if (occupiedRooms.size() == 0) {
+
+            } else {
+                combinedStream(occupiedRooms, unoccupiedRooms);
+            }
+            rooms = new PriorityQueue<>(new ComparableRoom());
+            for (int i = 0; i < numOfRooms; i++) {
+                Room room = internalList.get(i);
+                rooms.add(room);
+            }
+
+            int size = internalList.size();
+            for (int i = numOfRooms; i < size; i++) {
+                internalList.remove(numOfRooms);
+            }
+        }
+    }
+
+    /**
+     * Gives list of empty rooms in the region which exists after shrinkage.
+     */
+    public List<Room> unOccupiedRooms() {
```

Comment:
Mmm just wondering what this function and occupiedRooms() function is for? Do we need to know the list of empty and occupied rooms? Unless it's to find the first unoccupied room which I think your priority queue will have been able to find?